### ISSUE TITLE
Cannot install fluentd s3 plugin v1.8.0

### ISSUE THREAD

**Issue by Dimitris-Mil** [fluent/fluent-plugin-s3#444]
### Describe the bug

Error in installing fluentd s3 plugin. I get the following log while trying to install version 1.8.0

```
1 gem installed
Building native extensions. This could take a while...
ERROR:  Error installing fluent-plugin-s3:
	ERROR: Failed to build gem native extension.
    current directory: /fluentd/vendor/bundle/ruby/3.1.0/gems/zstd-ruby-1.5.6.6/ext/zstdruby
/usr/local/bin/ruby -I /usr/local/lib/ruby/3.1.0 extconf.rb
checking for rb_gc_mark_movable()... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.
Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/usr/local/bin/$(RUBY_BASE_NAME)
/usr/local/lib/ruby/3.1.0/mkmf.rb:490:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:583:in `try_link0'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:601:in `try_link'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:812:in `try_func'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:1116:in `block in have_func'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:989:in `block in checking_for'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:354:in `block (2 levels) in postpone'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:324:in `open'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:354:in `block in postpone'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:324:in `open'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:350:in `postpone'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:988:in `checking_for'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:1115:in `have_func'
	from extconf.rb:3:in `<main>'
To see why this extension failed to compile, please check the mkmf.log which can be found here:
  /fluentd/vendor/bundle/ruby/3.1.0/extensions/x86_64-linux/3.1.0/zstd-ruby-1.5.6.6/mkmf.log
extconf failed, exit code 1
Gem files will remain installed in /fluentd/vendor/bundle/ruby/3.1.0/gems/zstd-ruby-1.5.6.6 for inspection.
Results logged to /fluentd/vendor/bundle/ruby/3.1.0/extensions/x86_64-linux/3.1.0/zstd-ruby-1.5.6.6/gem_make.out
```

When specified the version 1.7.2 this plugin could be installed

### To Reproduce

Used this [helm chart](https://github.com/fluent/helm-charts/tree/main/charts/fluentd) to deploy fluentd 

and added the following plugins to be installed in the values.yaml

```yaml
plugins:
    - fluent-plugin-s3
```

### Expected behavior

I expected the plugin to be installed 

### Your Environment

```markdown
- Fluentd version: fluent/fluentd-kubernetes-daemonset:v1.16.2-debian-elasticsearch7-1.0
- TD Agent version: 
- fluent-plugin-s3 version: v1.8.0
- aws-sdk-s3 version:
- aws-sdk-sqs version:
- Operating system: linux
- Kernel version:
```

### Your Configuration

```yaml
plugins:
    - fluent-plugin-grafana-loki
    - fluent-plugin-s3 -v 1.8.0
    - fluent-plugin-forest
    - fluent-plugin-rewrite-tag-filter
```

### Your Error Log

```shell
1 gem installed
Building native extensions. This could take a while...
ERROR:  Error installing fluent-plugin-s3:
	ERROR: Failed to build gem native extension.
    current directory: /fluentd/vendor/bundle/ruby/3.1.0/gems/zstd-ruby-1.5.6.6/ext/zstdruby
/usr/local/bin/ruby -I /usr/local/lib/ruby/3.1.0 extconf.rb
checking for rb_gc_mark_movable()... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.
Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/usr/local/bin/$(RUBY_BASE_NAME)
/usr/local/lib/ruby/3.1.0/mkmf.rb:490:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:583:in `try_link0'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:601:in `try_link'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:812:in `try_func'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:1116:in `block in have_func'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:989:in `block in checking_for'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:354:in `block (2 levels) in postpone'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:324:in `open'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:354:in `block in postpone'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:324:in `open'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:350:in `postpone'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:988:in `checking_for'
	from /usr/local/lib/ruby/3.1.0/mkmf.rb:1115:in `have_func'
	from extconf.rb:3:in `<main>'
To see why this extension failed to compile, please check the mkmf.log which can be found here:
  /fluentd/vendor/bundle/ruby/3.1.0/extensions/x86_64-linux/3.1.0/zstd-ruby-1.5.6.6/mkmf.log
extconf failed, exit code 1
Gem files will remain installed in /fluentd/vendor/bundle/ruby/3.1.0/gems/zstd-ruby-1.5.6.6 for inspection.
Results logged to /fluentd/vendor/bundle/ruby/3.1.0/extensions/x86_64-linux/3.1.0/zstd-ruby-1.5.6.6/gem_make.out
```
```


### Additional context

_No response_

**Comment by Dimitris-Mil** [comment]
I've decided to use static versions for the plugins, so I added to the fluentd helm chart the following config.

```
plugins:
  - fluent-plugin-grafana-loki -v 1.2.20
  - fluent-plugin-s3 **-v 1.7.2**
  - fluent-plugin-forest -v 0.3.3
  - fluent-plugin-rewrite-tag-filter -v 2.4.0
```

This basically fixed the issue, since I used the 1.7.2 that worked before

**Comment by daipom** [comment]
Thanks for your report!
We should remove `zstd-ruby` dependency.
(It is not mandatory. we need `zstd-ruby` only when we want to use `zstd` compress type.)
We will fix this and release 1.8.1.

**Comment by kenhys** [comment]
fluent-plugin-s3 1.8.1 was released.
